### PR TITLE
[DOCU-2582] Add single source docs to the Contributing guide

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Summary

Added our internal docs about the single sourced versions and single sourced plugins to the external contributing guide.
### Reason

This information is helpful to all contributors, and it was kind of hidden/not in the place contributors would expect to find it. This PR makes it easier to find. 
### Testing

See the single sourced versions and single sourced plugins pages in the contributor's guide (they're in the Markdown rules and formatting section). 

Notes for reviewers:
- I only lightly edited these for the sake of getting them published ASAP. Is there any other reformatting that I should do for this initial PR?
- Is there any other tech information that is missing from these? @Guaris, I know you mentioned conditional rendering based on product type, but I don't remember how that was done.
- I thought about pulling the "Conditional rendering" section out into it's own page since it's related to single sourcing, but feels like there's enough content and it's different enough/used often enough, that it would help to have a separate page. Thoughts?

### CODE REVIEW

**Comment by lena-larionova** [comment]
Can you add https://konghq.atlassian.net/browse/DOCU-2582 to the title/description?

File: app/contributing/single-sourced-versions.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+| Test2 | Conditional | Hidden |
+{% endif_version %}
+
+| Test1 | Works       | Shows  |
+```
+{% endraw %}
+
+The above will be rendered as a single table.
+
+If you want to conditionalize content within a sentence, you can use `if_version` and specify `inline:true`:
```
Comment (cloudjumpercat) [inline app/contributing/single-sourced-versions.md:197]:
```suggestion
If you want to conditionally render content in a sentence, you can use `if_version` and specify `inline:true`:
```

File: app/contributing/single-sourced-versions.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+{% raw %}
+```
+Hello {% if_version eq:1.0.0 inline:true %}everyone in the {% endif_version %} world.
+```
+{% endraw %}
+
+### Conditionalize front matter
```
Comment (cloudjumpercat) [inline app/contributing/single-sourced-versions.md:205]:
```suggestion
### Conditionally render front matter
```

**Comment by Guaris** [comment]
@cloudjumpercat 


product conditional rendering looks like this
```
{% unless page.edition == "konnect" %}
This will be hidden for Konnect
{% endunless %}
```

I think the docs set would be stronger if conditional rendering was split out into it's own doc like you mentioned.

File: app/contributing/plugin-docs.md [hunk 3]
```diff
@@ -1,87 +1,70 @@
...
+1. Create a subdirectory for your plugin within your publisher directory.
+For example, `_app/_hub/kong-inc/your-plugin`.
+
+1. Copy the `/app/_hub/_init/my-extension/_index.md` and 
+the `/app/_hub/_init/my-extension/versions.yml` files into your plugin's subdirectory.
+
+1. Edit your `_index.md` file based on the guidance in comments in that file.
+You'll also find lots of helpful examples in other plugin doc files.
+
+1. Edit your `versions.yml` file with the minimum Kong Gateway version that this plugin supports.
```
Comment (lena-larionova) [inline app/contributing/plugin-docs.md:34]:
```suggestion
1. Edit your `versions.yml` file with the minimum {{site.base_gateway}} version that this plugin supports.
```

**Review by lena-larionova — APPROVED** [review]
I did a bunch of refactoring & editing of the single-sourcing topics after all. Here's a summary:
- Moved the conditional rendering info from the plugin single-sourcing doc into the separate topic
- Formatting, editing, general cleanup
- Set `no_version` as the default for all contributing guide pages. We won't have to set `no_version` manually on individual pages. 
- Updated and published the hidden plugin docs page
- Removed the duplicate single-sourcing docs from the /docs directory and linked off to the published pages instead.

**Pull request merged**